### PR TITLE
Add python bindings for Vec3 operators and constructors for NcsOp/Transform

### PR DESCRIPTION
**Pull request by keitaroyam**

### CODE REVIEW

File: python/unitcell.cpp [hunk 1]
```diff
@@ -123,6 +134,12 @@ void add_unitcell(py::module& m) {
 
   py::class_<Transform>(m, "Transform")
     .def(py::init<>())
+    .def(py::init([](const Mat33& m, const Vec3& v) {
+      Transform* tr = new Transform();
+      tr->mat = m;
+      tr->vec = v;
+      return tr;
```
Comment (sizmailov) [inline python/unitcell.cpp:141]:
```suggestion
      return Transform{m, v};
```

File: python/meta.cpp [hunk 2]
```diff
@@ -73,6 +73,13 @@ void add_meta(py::module& m) {
 
   py::class_<NcsOp>(m, "NcsOp")
     .def(py::init<>())
+    .def(py::init([](const Transform& tr, const std::string& id, bool given) {
+      NcsOp* op = new NcsOp();
+      op->tr = tr;
+      op->id = id;
+      op->given = given;
+      return op;
```
Comment (sizmailov) [inline python/meta.cpp:81]:
```suggestion
      return NcsOp{id, given, tr};
```

**Review by sizmailov — COMMENTED** [review]
Looks good to me. I think here we can follow Core Guidelines rules and [avoid naked new](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es60-avoid-new-and-delete-outside-resource-management-functions) by replacing 

```c++
T* t = new T();
... 
return t;
```
with 
```c++
T t;
... 
return t;
```
In the examples below we can go further and shorten lambda bodies to returns of inplace constructed structs.

**Comment by wojdyr** [comment]
@sizmailov pybind11::init() expects a function returning raw pointer. I think that if you return a value it then must be copied, so raw pointer is more efficient. It still could be in one line (`return new Transform{m, v};`), and I'd probably write it this way, but it's only a matter of personal preference.

**Comment by keitaroyam** [comment]
@sizmailov That's good to know! Thanks.

**Comment by sizmailov** [comment]
> pybind11::init() expects a function returning raw pointer. 

I think it is _possible_ to bind return-by-value constructors (just like any other function) from very early pybind releases. ~~With RVO and copy elision the impact on performance should be hardly measurable if any.~~ After writing that I headed to measuring some trivial example and I think I can see *tiny* (<0.5%), but still reproducible speed up in returning `new T(...)` over `T(...)` or `std::make_unique<T>(...)`, so you are right. Not sure where difference comes from in case of `make_unique` compared to bare pointer since pybind need to create it one way or another...

Anyhow there is no bug in the code, and comments are just about code style and it's up to you to decide how should it be shaped :)

**Comment by wojdyr** [comment]
I was also hoping that RVO and copy elision would remove all needless copying.
But then I noticed that python function gemmi.read_structure does a lot of memcpy, and changing it from:

    return read_structure(...);

to:

    return new Structure(read_structure(...));

removed the copying (so the copying got elided only in the second case). I suppose gemmi has more places that could be optimized.

**Pull request merged**